rts-core/src/components/game.rs

```diff
@@ -42,33 +60,37 @@ impl Game {
     }
 
     pub fn console_display(&self) -> Result<(), RtsException> {
-        let play_ground_ptdr = Arc::clone(&self.map);
-        let play_ground_mutex = play_ground_ptdr
-            .lock()
-            .map_err(|_| RtsException::GeneralException("".to_string()))?;
-        ConsoleDisplayer::display(&play_ground_mutex)
+        let play_ground_ptr = Rc::clone(&self.map);
+        let play_ground = play_ground_ptr.borrow();
+        ConsoleDisplayer::display(&play_ground)
     }
 
     /// Events loop to handle game state
-    pub async fn start(&self) -> Result<(), RtsException> {
+    pub fn start(&self) -> Result<(), RtsException> {
         loop {
             self.execute_recurring_actions()?;
-
+            self.play_with_all_players()?;
+            self.update_observers()?;
             if self.check_game_is_over()? {
                 break;
             }
-            time::sleep(std::time::Duration::from_secs(10)).await;
+
+            thread::sleep(Duration::from_secs(TURN_DURATION_IN_SECONDS));
         }
         Ok(())
     }
 
-    /// Plays a turn with player execute given action
-    pub async fn play_with_async(&self, index: usize, action: Action) -> Result<(), RtsException> {
-        self.play_with(index, action)
+    fn play_with_all_players(&self) -> Result<(), RtsException> {
+        for (i, player) in self.players.iter().enumerate() {
+            let player_ptr = Rc::clone(player);
+            let action = player_ptr.borrow().request()?;
+            self.play(i, action)?;
```

Comment:
If I understand this correctly, a player can only play one single action in a turn ?

Reply:
Yup, maybe we will evolve this later. I guess this method should change with how we handle Actix actors

Reply:
Should we implement actors on this PR ?

Reply:
I think it should be fine as is, we probably don't need actors just yet (or at all?)